dace/frontend/common/distr.py

```diff
@@ -45,6 +187,44 @@ def _bcast(pv: ProgramVisitor,
     return None
 
 
+@oprepo.replaces_method('Cartcomm', 'Bcast')
+@oprepo.replaces_method('Intracomm', 'Bcast')
+def _intracomm_bcast(pv: 'ProgramVisitor',
+                     sdfg: SDFG,
+                     state: SDFGState,
+                     comm: Tuple[str, 'Comm'],
+                     buffer: str,
+                     root: Union[str, sp.Expr, Number] = 0):
+    """ Equivalent to `dace.comm.Bcast(buffer, root)`. """
+
+    from mpi4py import MPI
+    comm_name, comm_obj = comm
+    if comm_obj == MPI.COMM_WORLD:
+        return _bcast(pv, sdfg, state, buffer, root)
+    # NOTE: Highly experimental
```

Comment:
is it?

Reply:
Yes, because it currently requires the user to explicitly pass the Fortran ID or object's address to the program's call. I think a better solution would utilize the program's closure with some custom handling to automate this process. This needs a bit of extra design time.